Cython/CodeWriter.py

```diff
@@ -245,10 +249,43 @@ def visit_FuncDefNode(self, node):
         self.visit(node.body)
         self.dedent()
 
+    def visit_CFuncDefNode(self, node):
+        if node.overridable:
+            self.startline(u'cpdef ')
+        else:
+            self.startline(u'cdef ')
+        if node.modifiers:
+            self.put(' '.join(node.modifiers))
+            self.put(' ')
+        if node.visibility != 'private':
+            self.put(node.visibility)
+            self.put(u' ')
+        if node.api:
+            self.put(u'api ')
+
+        if node.base_type:
+            self.visit(node.base_type)
+            if node.base_type.name is not None:
+                self.put(u' ')
+
+        # visit the CFuncDeclaratorNode, but put a `:` at the end of line
+        self.visit(node.declarator.base)
+        self.put(u'(')
+        self.comma_separated_list(node.declarator.args)
+        self.endline(u'):')
+
+        self.indent()
+        self.visit(node.body)
+        self.dedent()
```

Comment:
This pattern seems worth wrapping in a `visit_indented()` method.

Reply:
Done.